system_tests/system_tests_sync/test_external_accounts.py

```diff
@@ -211,3 +230,54 @@ def __enter__(self):
                 },
             },
         )
+
+
+def test_aws_based_external_account(
+    aws_credentials, service_account_info, dns_access, http_request
+):
+
+    response = http_request(
+        url=(
+            "https://sts.amazonaws.com/"
+            "?Action=AssumeRoleWithWebIdentity"
+            "&Version=2011-06-15"
+            "&DurationSeconds=3600"
+            "&RoleSessionName=python-test"
+            "&RoleArn={}"
+            "&WebIdentityToken={}"
+        ).format(_ROLE_AWS, aws_credentials)
+    )
+    assert response.status == 200
+
+    # The returned data is in XML, but loading an XML parser would be overkill.
+    # Searching the return text manually for the start and finish tag.
+    data = response.data.decode("utf-8")
+
+    def get_xml_value_by_tagname(tagname):
+        startIndex = data.index("<{}>".format(tagname))
+        if startIndex >= 0:
+            endIndex = data.index("</{}>".format(tagname), startIndex)
+            if endIndex > startIndex:
+                return data[startIndex + len(tagname) + 2 : endIndex]
+
+    os.environ["AWS_REGION"] = "us-east-2"
```

Comment:
Should we use `with patch.dict(os.environ` to populate these as it takes care of cleanup?

Reply:
Done